Framework/Algorithms/src/RadiusSum.cpp

```diff
@@ -619,16 +613,16 @@ void RadiusSum::setUpOutputWorkspace(std::vector<double> &values) {
       inputWS, 1, values.size() + 1, values.size());
 
   g_log.debug() << "Set the data\n";
-  MantidVec &refY = outputWS->dataY(0);
-  std::copy(values.begin(), values.end(), refY.begin());
+  outputWS->mutableY(0) = values;
 
   g_log.debug() << "Set the bins limits\n";
-  MantidVec &refX = outputWS->dataX(0);
+
+  auto xSize = outputWS->mutableX(0).size();
   double bin_size = (max_radius - min_radius) / num_bins;
+  outputWS->setBinEdges(0, xSize,
+                        HistogramData::LinearGenerator(min_radius, bin_size));
 
-  for (int i = 0; i < (static_cast<int>(refX.size())) - 1; i++)
-    refX[i] = min_radius + i * bin_size;
-  refX.back() = max_radius;
+  outputWS->mutableX(0).back() = max_radius;
```

Comment:
Well, I think the reason it is there is to avoid floating point inaccuracies. With exact arithmetic it would make no difference. In finite precision there will be a very small difference. In my opinion difference is negligible (as confirmed by your tests) --- if this finite precision error would make a difference anywhere it would indicate a much larger issue somewhere.

So I'd remove it.